src/App.js

```diff
@@ -20,11 +28,10 @@ function App() {
         </Route>
 
         <Route path="/">
-          <Welcome />
+          {token ? <Redirect to="/list" /> : <Welcome onClick={handleClick} />}
```

Comment:
This will redirect to `list` if the initial default value of `token` is a token from `localStorage`

Reply:
The rendering for the Welcome component is inside the "/" route, so if a user navigates directly to Add Items or List there is potential that the user token is not already set in localStorage.  So I would suggest that we might want to render Welcome instead of going through the routes logic when no user token exists.

Reply:
The rendering for the Welcome component is inside the "/" route, so if a user navigates directly to Add Items or List there is potential that the user token is not already set in localStorage. So I would suggest that we might want to render Welcome instead of going through the routes logic when no user token exists.

Reply:
If we just render `<Welcome />`, then would we do something like `{ token && <Redirect to="/list" />"` underneath the `<Welcome />` component or could the `<Redirect /> be anywhere within the `App` component? Or is the suggestion that the logic for managing the `token` state and generating the token and saving it to `localStorage` should just be inside of the `<Welcome />` component? I just want to make sure I'm on the same page! 👍

Reply:
I'm thinking that we should not render the components that require a user token until one exists.  The logic that renders the components that depend on user token are handled by the Switch component, so do not render Switch component until the user has a token.

Reply:
Okay! We've implemented those changes. Yay team!